### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Rustamxon7**
### TDD project

**Create methods for:**

- `factorial`
- `reverse`
- `fizzbuzz`

Feel free to review🧑🏻‍💻 our project and let us know if there are any changes to do!

### CODE REVIEW

File: spec/solver_test.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  describe '#factorial' do
+    it 'returns 1 when given 0' do
+      expect(@solver.factorial(0)).to eq(1)
+    end
+
+    it 'returns 120 when given 5' do
+      expect(@solver.factorial(5)).to eq(120)
+    end
+  end
+
```
Comment (Mihndim2020) [inline spec/solver_test.rb:18]: - [ ] As it is a requirement for this project and a best  practice to cover all the test cases, please kindly add a test for  the factorial of a `negative number`.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark: Your tests are running. 

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

- [ ] Following the rules of `TDD(test driven development)` which is also a requirement for this project, you are suppose to write the tests first  before implementing the `Solver class` and the various methods.  Your `commit history` should show that you implemented the tests first. So please first implement the tests and commit before you write the methods.

rspec

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omar-labana** [comment]
Hi team, 

Your project is complete! :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
